### PR TITLE
Settings selector

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
This pull request covers the controls under the "Trip Options & Preferences" tab in the OTP UI, including:

* Mode and submode selection,
* Company/provider selection
* Trip optimization parameters

### CODE REVIEW

**Comment by binh-dam-ibigroup** [comment]
I'm good with renaming the top package.
For individual components, how about I combine the stories for the UI sub-components?

> Hey Bihn,
> 
> 2 thoughts:
> a) can we rename the top level directory from settings-selector to trip-form, or trip-form-components?
> b) should we have 1 storybook as opposed to multiple stories? (debate can be made either way...I like that all the separate components are under a single package/ , but no loving the multiple stories).

**Comment by fpurcell** [comment]
+1

**Comment by binh-dam-ibigroup** [comment]
@fpurcell It is the name that was used in the old code...

> This is great, Binh. I like the rename and the storybook layout and the functionality of the SettingsSelectorPanel (not crazy about the name "SettingsSelector", but that's a minor quibble since the code is under the trip-form bucket).

File: packages/trip-form/src/ModeIcon/index.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import React from "react";
+import PropTypes from "prop-types";
+import * as Icons from "@opentripplanner/icons";
+
+/**
+ * A generic icon component that displays an icon for the specified transportation mode.
+ */
```
Comment (landonreed) [inline packages/trip-form/src/ModeIcon/index.js]: Is this duplicative of https://github.com/opentripplanner/otp-ui/blob/master/packages/icons/src/trimet-mode-icon.js?

File: packages/trip-form/src/ModeSelector/index.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+  const handleClick = option => {
+    if (!option.selected && typeof onChange === "function") {
+      onChange(option.id);
+    }
+  };
+
+  const makeButton = option => (
+    <ModeButton
+      /* If key is missing, a warning in React Dev Tools is triggered. */
```
Comment (landonreed) [inline packages/trip-form/src/ModeSelector/index.js]: This comment probably isn't needed. That's a relatively well known prop/warning.

File: packages/trip-form/src/SubmodeSelector/index.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  const RowType = inline ? InlineSubmodeRow : SubmodeRow;
+
+  return (
+    <SettingsSection className={className} style={style}>
+      {label && <LabelType>{label}</LabelType>}
+      <RowType>
+        {modes &&
+          modes.map(option => (
+            <ModeButton
+              /* If key is missing, a warning in React Dev Tools is triggered. */
```
Comment (landonreed) [inline packages/trip-form/src/SubmodeSelector/index.js]: Again, comment probably not needed

File: packages/trip-form/src/ModeIcon/index.js (hunk 1 above)
Comment (binh-dam-ibigroup) [reply packages/trip-form/src/ModeIcon/index.js]: I didn't see that one... the original code I based this work was referring to this file instead: https://github.com/opentripplanner/otp-react-redux/blob/dev/lib/components/icons/mode-icon.js

If creating an issue, is it against this repo or against otp-react-redux?

Comment (landonreed) [reply packages/trip-form/src/ModeIcon/index.js]: Probably this repo is OK.

File: packages/trip-form/src/GeneralSettingsPanel/index.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+import queryParams from "@opentripplanner/core-utils/lib/query-params";
+import {
+  defaultParams,
+  getQueryParamProperty
+} from "@opentripplanner/core-utils/lib/query";
+import { queryType } from "@opentripplanner/core-utils/lib/types";
+
+import CheckboxSelector from "../CheckboxSelector";
+import DropdownSelector from "../DropdownSelector";
+import commonModes from "../__mocks__/modes"; // FIXME: Replace with ref to configuration.
```
Comment (landonreed) [inline packages/trip-form/src/GeneralSettingsPanel/index.js]: I don't think we should be importing from `__mocks__` in the files that make it to the library (i.e., not test or story files). My recommendation would be to make this a constant that is imported from a `consts` file and set this as a default value for a new prop `modes`. I'm not crazy about the `configWrapper` below, especially if all that's needed is the modes field.

Comment (binh-dam-ibigroup) [reply packages/trip-form/src/GeneralSettingsPanel/index.js]: Oh good catch, I forgot about getting rid of this one.

File: packages/trip-form/src/__mocks__/modes.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      mode: "RAIL",
+      label: "WES"
+    },
+    {
+      mode: "GONDOLA",
+      label: "Aerial Tram"
+    }
+  ],
+  accessModes: [
+    // Biketown and Uber should be displayed the same despite Biketown company set in companies.
```
Comment (landonreed) [inline packages/trip-form/src/__mocks__/modes.js]: This comment reads a bit cryptic, but seems important. Could you elaborate here?

File: packages/core-utils/src/state.js [hunk 6]
```diff
@@ -0,0 +1,151 @@
+import isEqual from "lodash.isequal";
```
Comment (landonreed) [inline packages/core-utils/src/state.js]: Is this `state.js` file used anywhere?

Comment (binh-dam-ibigroup) [reply packages/core-utils/src/state.js]: Yes, query.js has this:
`import { getActiveSearch } from "./state";`

File: packages/trip-form/src/__mocks__/modes.js (hunk 5 above)
Comment (binh-dam-ibigroup) [reply packages/trip-form/src/__mocks__/modes.js]: Yep. Fixed in lastest commit.

File: packages/core-utils/src/state.js (hunk 6 above)
Comment (landonreed) [reply packages/core-utils/src/state.js]: Oh, I was just confused because I didn't see any imports in this PR.

**Pull request merged**